### DISCUSSION TITLE
Infinite scroll

### DISCUSSION THREAD

**Discussion by saulfrr** [typesense/typesense-swift#20] [General]
Hi, I have been testing the library and I can get hits from TypeSense server. Now, is there a way in the library to implement infinite scroll? 

Regards.

**Comment by programVeins** [comment]
Hey there! Typesense server takes care of paginations for you if you need them. You can implement them in swift by adding the `page` and `perPage` search parameters to your SearchParameters object. [Reference](https://github.com/typesense/typesense-swift/blob/9fa96470233661287a574220db28aa99c2ff2e10/Sources/Typesense/Models/SearchParameters.swift#L36-L39)

You can now use these as limit and offset to fetch more results at the end of scroll. Eg: set `perPage` to a constant (say 10) and increment `page` once we hit bottom of list.

**Comment by saulfrr** [comment]
Hi,

Yes, that is how I did it. Thanks.

         let results = data?.results[0]
                
                if let pageNumber = results?.page {
                      self.nextPageNumber = pageNumber + 1
                }

Now, in the SwiftUI version of the example, when you make a search and get some hits, you are able to scroll down through the results. In the UIKit version the scroll is stuck after you get some hits back.

Regards.

**Reply by saulfrr** [reply]
Ok, in ViewController.swift I used:

`let searchController = UISearchController(searchResultsController: nil)`

Instead of:

`let searchController = UISearchController(searchResultsController: HitsViewController())`

And now it is able to scroll after getting some hits back.
